orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/EvaluateConditionTask.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.tasks;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.conditions.ConditionConfigurationProperties;
+import com.netflix.spinnaker.orca.conditions.Condition;
+import com.netflix.spinnaker.orca.conditions.ConditionSupplier;
+import com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.time.Clock;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.netflix.spinnaker.orca.pipeline.WaitForConditionStage.WaitForConditionContext.*;
+
+@Component
+public class EvaluateConditionTask implements RetryableTask {
+  private static Logger log = LoggerFactory.getLogger(EvaluateConditionTask.class);
+  private final ConditionConfigurationProperties conditionsConfigurationProperties;
+  private final Optional<List<ConditionSupplier>> suppliers;
+  private final Clock clock;
+
+  @Autowired
+  public EvaluateConditionTask(
+    ConditionConfigurationProperties conditionsConfigurationProperties,
+    Optional<List<ConditionSupplier>> suppliers,
+    Clock clock
+  ) {
+    this.conditionsConfigurationProperties = conditionsConfigurationProperties;
+    this.suppliers = suppliers;
+    this.clock = clock;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.MINUTES.toMillis(conditionsConfigurationProperties.getBackoffWaitMins());
```

Comment:
Thinking about this more: The backoff should be much smaller so to allow detecting a skipped execution.
I am going to update this so that the backoff is explicitly checked in the task execution instead.